robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+
+        :returns dict: The dict of varios defect types and their codes
+        """
+        _defect_types = {
+            'product_bug': 'PB001',
+            'automation_bug': 'AB001',
+            'system_issue': 'SI001',
+            'to_investigate': 'TI001',
+            'no_defect': 'ND001',
+        }
+        return _defect_types
+
+    def _params(self, **kwargs):
+        """Parameters for Test items API request
+
+        :param dict kwargs: The tests `defect_type` and/or testss `status`
+
+        It contains filters to get the test items data from launch
+
+        :returns dict: parameters for API test items request
+        """
+        params = [
+            ('filter.eq.launch', self.launch['id']),
+            ('page.page', 1),
+            ('page.size', 50),
+            ('page.sort', 'start_time'),
+        ]
+        defect_type = kwargs.get('defect_type')
+        test_status = kwargs.get('status')
+        if defect_type and defect_type in self.defect_types:
+            params.insert(0, ('filter.eq.issue$issue_type', self.defect_types[defect_type]))
+        if test_status:
+            params.insert(2, ('filter.eq.status', test_status))
+        return dict(params)
+
+    @retry(
+        stop=stop_after_attempt(4), wait=wait_fixed(10),
+    )
+    def _requester(self, params, page):
+        """The Test Items GET requester to fetch the data on a page
+
+        If any error and before Failing explicitly, it retries for 3 times with 10 seconds delay
+
+        :param dict params: The parameters dictionary for API request
+        :param int page: The Page number parameter manipulates the params
+
+        :returns tuple (int, list): Total pages count and the list of tests along with
+            each tests properties in a page
+        """
+        params['page.page'] = page
+        resp = requests.get(self.tests_url, headers=self.headers, params=params, verify=False)
+        resp.raise_for_status()
+        total_pages = resp.json()['page']['totalPages']
+        pagedata = resp.json().get('content')
+        return total_pages, pagedata
+
+    def _tests(self, params, data, page=1):
+        """The Recursive Function that roll over on all pages to get all tests data in an API request
+
+        :param dict params: The parameters dictionary for API request
+        :param list data: The Empty dict to be filled during recursion with all tests data
+        :param page: The initial page which will be changed to next during recursion
+
+        :returns dict: All filtered tests based on params data along with each tests properties
+        """
+        LOGGER.debug(f"Fetching results for Page : {page}")
+        total_pages, pagedata = self._requester(params=params, page=page)
+        data += pagedata
+        if page < total_pages:
+            page += 1
+            self._tests(params=params, data=data, page=page)
+        return data
+
+    def tests(self, **kwargs):
```

Comment:
`test_items`?

Reply:
same comment as above for making it compatible with robottelo terms.

Reply:
Potato/Potato - we'll be using this within pytest hooks, where the tests are just called `items`. Some tangential benefit to the method here being named a bit differently.